code/src/qik_2s12v10.cpp

```diff
@@ -2,30 +2,108 @@
 
 namespace r2d2::moving_platform {
 
+    hwlib::ostream &
+    operator<<(hwlib::ostream &out,
+               const qik_2s12v10_set_configuration_parameter_return &rhs) {
+        if (rhs == qik_2s12v10_set_configuration_parameter_return::command_ok) {
+            out << "command ok";
+        } else if (rhs == qik_2s12v10_set_configuration_parameter_return::
+                              bad_parameter) {
+            out << "bad parameter";
+        } else if (rhs ==
+                   qik_2s12v10_set_configuration_parameter_return::bad_value) {
+            out << "bad value";
+        } else {
+            out << "unknown return";
+        }
+        return out;
+    };
+
+    r2d2::usart::usart_connection_c &
+    operator<<(r2d2::usart::usart_connection_c &out,
+               const qik_2s12v10_registers &rhs) {
+        return out << static_cast<uint8_t>(rhs);
+    }
+    r2d2::usart::usart_connection_c &
+    operator<<(r2d2::usart::usart_connection_c &out,
+               const qik_2s12v10_configuration_parameter &rhs) {
+        return out << static_cast<uint8_t>(rhs);
+    }
+
+    hwlib::ostream &operator<<(hwlib::ostream &out,
+                               const qik_2s12v10_error &rhs) {
+        switch (rhs) {
+        case qik_2s12v10_error::motor_0_fault:
+            out << "Motor_0_fault";
+            break;
+        case qik_2s12v10_error::motor_1_fault:
+            out << "Motor_1_fault";
+            break;
+        case qik_2s12v10_error::motor_0_over_current:
+            out << "Motor_0_over_current";
+            break;
+        case qik_2s12v10_error::motor_1_over_current:
+            out << "Motor_1_over_current";
+            break;
+        case qik_2s12v10_error::serial_hardware_error:
+            out << "Serial_hardware_error";
+            break;
+        case qik_2s12v10_error::crc_error:
+            out << "Crc_error";
+            break;
+        case qik_2s12v10_error::format_error:
+            out << "Format_error";
+            break;
+        case qik_2s12v10_error::timeout:
+            out << "Timeout";
+            break;
+        case qik_2s12v10_error::no_error:
+            out << "No error";
+            break;
+        default:
+            out << "Unkown error";
+            break;
+        }
+        return out;
+    }
+
     qik_2s12v10_c::qik_2s12v10_c(r2d2::usart::usart_connection_c &usart_bus,
-                      hwlib::pin_out &reset_pin)
-        : reset_pin(reset_pin), 
-        usart_bus(usart_bus)  {
-            init();
-            get_configuration_parameter(1);
+                                 hwlib::pin_out &reset_pin)
+        : reset_pin(reset_pin), usart_bus(usart_bus) {
+        init();
+        hwlib::cout << "Device id: "
```

Comment:
Should this cout be here?

Reply:
It has been there in a way for a while but I will remove it.